File: go/pricing/pricesummary.go
// ABOUTME: PriceSummary domain type representing the final pricing result.
// ABOUTME: Contains subtotal, discount total, final total, and applied discounts.
package pricing

import "promotionengine/domain"

// PriceSummary represents the result of pricing a cart.
type PriceSummary struct {
	Subtotal         domain.Money
	DiscountTotal    domain.Money
	Total            domain.Money
	AppliedDiscounts []AppliedDiscount
}

// NewPriceSummary creates a new PriceSummary.
func NewPriceSummary(subtotal, discountTotal, total domain.Money, appliedDiscounts []AppliedDiscount) PriceSummary {
	return PriceSummary{
		Subtotal:         subtotal,
		DiscountTotal:    discountTotal,
		Total:            total,
		AppliedDiscounts: appliedDiscounts,
	}
}

File: typescript/src/pricing/PriceSummary.ts
// ABOUTME: Complete pricing breakdown for a cart
// ABOUTME: Contains subtotal, discount total, final total, and applied discounts

import { Money } from '../domain/Money';
import { AppliedDiscount } from './AppliedDiscount';

export class PriceSummary {
  private readonly _appliedDiscounts: readonly AppliedDiscount[];

  constructor(
    public readonly subtotal: Money,
    public readonly discountTotal: Money,
    public readonly total: Money,
    appliedDiscounts: AppliedDiscount[]
  ) {
    // Defensive copy and freeze
    this._appliedDiscounts = Object.freeze([...appliedDiscounts]);
  }

  get appliedDiscounts(): readonly AppliedDiscount[] {
    return this._appliedDiscounts;
  }
}
